crates/chain/src/keychain/txout_index.rs

```diff
@@ -206,9 +206,10 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
     fn next_store_index(&self, keychain: &K) -> u32 {
         self.inner()
             .all_spks()
-            .range((keychain.clone(), u32::MIN)..(keychain.clone(), u32::MAX))
+            .keys()
+            .filter(|(k, _)| k == keychain)
```

Comment:
I advise against using `filter` here. `filter` will iterate over every single spk whereas `range` only iterates over spks of the given keychain.

Additionally, `btree::Range` has an optimized implementation of `Iterator::last` (https://doc.rust-lang.org/src/alloc/collections/btree/map.rs.html#1981) which we will be missing out on if we use `filter`.

Reply:
Good catch! I added a comment explaining what is `range` doing and why we should use it